app/components/Editor.js

```diff
@@ -129,23 +132,39 @@ export default class Editor extends React.Component {
         $R('#editor', 'module.buffer.clear');
       },
       setEditorRawText: (rawText) => {
-        // Get the current caret location
-        let caretLocation = this.redactor.selection.getPosition();
-        let point = {clientX: caretLocation.left, clientY: caretLocation.top};
-
-        let cleaned = this.redactor.cleaner.input(rawText);
+        // Sets the code for the editor, maintains caret location if available.
+
+        let focused = this.redactor.editor.isFocus();
+        let point;
+
+        if (focused) {
+          // Attempt to save caret location, otherwise ignore.
+          try {
+            let caretLocation = this.redactor.selection.getPosition();
+            point = {clientX: caretLocation.left, clientY: caretLocation.top};
+          } catch (error) {
+            focused = false;
```

Comment:
Seems like this should be a readonly value. Perhaps it adds convenience not creating a new variable to track this state, but that's what we should do.

Reply:
It was initially a readonly value but after adding the try-catch I changed it to make it easier. I'll find a better way to do this.